spec/mailers/shipment_mailer_spec.rb

```diff
@@ -13,6 +13,20 @@
     allow(shipment).to receive_messages(tracking_url: "TRACK_ME")
     shipment
   end
+
+  let(:picked_up_shipment) do
+    order = build(:order_with_distributor)
+    product = build(:product, name: %{The "BEST" product})
+    variant = build(:variant, product: product)
+    line_item = build(:line_item, variant: variant, order: order, quantity: 1, price: 5)
+    shipment = build(:shipment)
+    allow(shipment).to receive_messages(line_items: [line_item], order: order)
+    # The only difference between a picked_up_email and shipped_email is that
+    # a picked_up email is sent for those shipments whose
+    # require_ship_address is false
+    allow(shipment).to receive_messages(require_ship_address: true)
```

Comment:
You should be able to set these variables on the shipment instead of mocking it. That's simpler and more realistic.